core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala

```diff
@@ -634,18 +634,43 @@ class ReassignPartitionsCommand(zkClient: KafkaZkClient,
         // Create reassignment znode so that controller will send LeaderAndIsrRequest to create replica in the broker
         zkClient.createPartitionReassignment(validPartitions.map({case (key, value) => (new TopicPartition(key.topic, key.partition), value)}).toMap)
 
-        // Send AlterReplicaLogDirsRequest again to make sure broker will start to move replica to the specified log directory.
-        // It may take some time for controller to create replica in the broker. Retry if the replica has not been created.
-        var remainingTimeMs = startTimeMs + timeoutMs - System.currentTimeMillis()
-        val replicasAssignedToFutureDir = mutable.Set.empty[TopicPartitionReplica]
-        while (remainingTimeMs > 0 && replicasAssignedToFutureDir.size < proposedReplicaAssignment.size) {
-          replicasAssignedToFutureDir ++= alterReplicaLogDirsIgnoreReplicaNotAvailable(
-            proposedReplicaAssignment.filter { case (replica, _) => !replicasAssignedToFutureDir.contains(replica) },
-            adminClientOpt.get, remainingTimeMs)
-          Thread.sleep(100)
-          remainingTimeMs = startTimeMs + timeoutMs - System.currentTimeMillis()
+        if (proposedReplicaAssignment.nonEmpty) {
+          // Send AlterReplicaLogDirsRequest again to make sure broker will start to move replica to the specified log directory.
+          // It may take some time for controller to create replica in the broker. Retry if the replica has not been created.
+          var remainingTimeMs = startTimeMs + timeoutMs - System.currentTimeMillis()
+          def nonFutureReplicas() = adminClientOpt.get
```

Comment:
Would it be reasonable to pull this change into a separate PR? I think the fix in the log dir fetcher is more important and a more isolated change will be easier to backport to older versions.

Reply:
Got it. trace this by https://issues.apache.org/jira/browse/KAFKA-9721